Playback OSD files in DVR

Knifa opened the pull request:
Let's get this PR up as a draft since it's nearly done!

Roughly:
- Further hook `duss_osal_task_create` to grab `player_thread` and `vdec_thread` to steal contexts. These have the state (i.e., pause status, current playback time, etc.) we need. These threads get recreated every time a new file gets played.
- When playback is detected (i.e., thread contexts are created and indicate playing), try load the matching OSD files, fonts, etc., etc., then bust out into a playback-specific loop for the duration.
- In that new loop, tick through frames, etc., etc., rendering them as if they were regular OSD frames.
- Undo all of the above when playback stops.

Pausing and seeking works as you would expect. Frame timing is eh... since it only updates `frames_sent`, `play_tm_ms`, etc. when it sends new frames, which it does in chunks as the RTOS needs it, but good enough for emergency viewing. The RTOS itself is handling putting out frames.

Gated under `rec_pb_enabled` just like recording since it feels sketchy enough, still.

Main thing I'm thinking about at this stage is, if I should integrate it into the main event loop instead --- like I don't know what happens if you try to playback DVR yet while you've got a live feed coming in. I guess we still want to be processing them? Need to pass out some builds for early testing, too.

Commit:
Enable OSD recording

Commit:
Redef/ordering fixup

Commit:
drop unneeded formatting

Commit:
First pass

Commit:
fix crash on subsequent plays

Commit:
misc tidy up

Commit:
only attempt playback once per session

Commit:
fix typo

Commit:
working seek, improv. timing

Commit:
tidy up

Commit:
fix is playing check

Commit:
handle pause

Commit:
formatting

Commit:
fix debug typo

Commit:
pb enable flag

Commit:
fix pause logic again...

Commit:
fix warning

Commit:
clear screen on leave

bri3d commented:
> main thing I'm thinking about at this stage is, if I should integrate it into the main event loop instead --- like I don't know what happens if you try to playback DVR yet while you've got a live feed coming in. 

I wouldn't think too hard about this, it should be similar to or the same as a connection drop, which I test against and should work fine. Besides the HD/SD switch in non-compressed mode, the goggles side should be stateless so the next packet that comes in should redraw the whole display. 

This _is_ a good incentive to make compressed mode the eternal default finally and consider ripping out the non-compressed mode code eventually.

Commit:
fix schema indent

Commit:
smol tidy up

Commit:
drop frame interp, less of an issue at 15hz

Knifa commented:
Nice, I did just give it a quick test (SD only though) and it seems to work fine.

I dropped the frame interp. since it's less of an issue now I dropped the frame rate (and also it kept running ahead when I looked into it more) --- think it's easier just to accept some hitches given it's for emergencies mainly.

Think this is ready for a review. 👍

bri3d commented:
Looks good to me I think. The feature switch seems pretty robust so I guess we'll just :shipit: and then see what happens
